### ISSUE TITLE
Touching entries in sidebars is broken

### ISSUE THREAD

**Issue by paper42** [skyjake/lagrange#298]
Touching an entry in bookmarks selects an entry shifted up or down a few entries.
On the feeds sidebar, touching anything selects the first entry.
On the history sidebar, touching an entry either selects the last selected one (first, then second, etc.) or doesn't do anything.

tested on 1.5.0

**Comment by skyjake** [comment]
Sounds like the touch events are being applied to obsolete coordinates... There may be some confusion whether SDL or my own code is tracking the "mouse" position for clicks when doing touch events. I'll look around a bit.

FWIW, I'm not seeing this behavior on iOS, so perhaps some touch state updates are currently dependent on `iPlatformAppleMobile` or `iPlatformMobile` flags.

**Comment by skyjake** [comment]
For debugging purposes, I tried converting all mouse events to touch events and running this on a desktop. Unfortunately, I could not reproduce the behavior you describe.

Is the problem isolated to the sidebar(s) or are touch events misplaced elsewhere in the UI?

**Comment by skyjake** [comment]
This may be helpful, though: in src/ui/window.c, there is a function called `mouseCoord_Window()`. Could you try enabling the first ifdef-branch to be executed always? So it looks like:

```c
iInt2 mouseCoord_Window(const iWindow *d) {
    iUnused(d);
    return latestPosition_Touch();
}
```

If that fixes the behavior, I'll need to modify the event processing so that this function queries Touch for the coordinates whenever the event being processed is actually a touch event and not a mouse event.

**Comment by paper42** [comment]
It seems to be a bit better now (1.5.2), touches are not misplaced anymore (not sure what changed), but it still doesn't always activate the entries - I can see the entry being highlighted, but nothing happens when touching it.

**Comment by paper42** [comment]
https://user-images.githubusercontent.com/23639164/123415134-1d29af00-d5b5-11eb-8c01-57447c0739f0.mp4

**Comment by skyjake** [comment]
That's the behavior with unmodified v1.5.2?

**Comment by paper42** [comment]
yes

**Comment by paper42** [comment]
This seems to be fixed in 1.6.0
